neural_compressor/torch/algorithms/weight_only/save_load.py

```diff
@@ -38,14 +38,472 @@ def save(model, output_dir="./saved_results"):
+        return model
+    else:
+        raise ValueError("`format` in load function can only be 'huggingface' or 'default', but get {}".format(format))
+
+
+def _build_woq_model(model, quantization_config, loaded_state_dict_keys):
+    """Build weight-only quantization model."""
+    from neural_compressor.torch.utils import set_module
+
+    from .modules import MulLinear, WeightOnlyLinear
+
+    for name, module in model.named_modules():
+        # get quantization config of module
+        module_name_type = str((name, type(module).__name__))
+        module_quantization_config = quantization_config
+        if module_name_type in quantization_config:
+            module_quantization_config = [config for config in quantization_config[module_name_type].values()][0]
+
+        if isinstance(module, torch.nn.Linear):
+            # module without qweight means it is not quantized, then skip it
+            loaded_state_dict_keys_set = set(loaded_state_dict_keys)
+            if (
+                name + ".qweight" not in loaded_state_dict_keys_set
+                and name + ".linear.qweight" not in loaded_state_dict_keys_set
+            ):
+                continue
+
+            # insert MulLinear module
+            if name + ".linear.qweight" in loaded_state_dict_keys_set:
+                new_module = MulLinear(module)
+                set_module(model, name, new_module)
+                name += ".linear"
+
+            # replace `torch.nn.Linear` with `WeightOnlyLinear`
+            zp = True if name + ".qzeros" in loaded_state_dict_keys_set else False
+            g_idx = True if name + ".g_idx" in loaded_state_dict_keys_set else False
+            new_module = WeightOnlyLinear(
+                module.in_features,
+                module.out_features,
+                dtype=module_quantization_config.get("dtype", "int"),
+                bits=module_quantization_config.get("bits", 4),
+                group_size=module_quantization_config.get("group_size", 32),
+                zp=zp,
+                bias=module.bias is not None,
+                g_idx=g_idx,
+                use_optimum_format=True,
+            )
+            set_module(model, name, new_module)
+
+    return model
+
+
+def _load_inc_woq_model(qmodel_weight_file_path, qconfig_file_path, origin_model):
+    qweights = torch.load(qmodel_weight_file_path)
+
+    quantization_config = {}
+    with open(qconfig_file_path, "r") as file:
+        quantization_config = json.load(file)
+
+    model = _build_woq_model(origin_model, quantization_config, qweights.keys())
+    model.load_state_dict(qweights, assign=True)
+    model.eval()
+    return model
+
+
+def _load_hf_woq_model(pretrained_model_name_or_path, *model_args, **kwargs):
+    # check required package
+    try:
+        import transformers
+    except ImportError:
+        logger.error("`transformers` package is required for loading hugginface weight-only quantization model.")
```

Comment:
This exception handling will log the error and continue the execution. I guest the expected behavior is to attempt to install the required package automatically. If failed, raise an import error.